manticore/core/cpu/arm.py

```diff
@@ -332,20 +335,20 @@ def __getstate__(self):
     def __setstate__(self, state):
         super(Armv7Cpu, self).__setstate__(state)
         self._last_flags = state['_last_flags']
-        self._at_symbolic_conditional = state['at_symbolic_conditional'] 
+        self._at_symbolic_conditional = state['at_symbolic_conditional']
 
     def _set_mode(self, new_mode):
-        assert new_mode in (CS_MODE_ARM, CS_MODE_THUMB)
+        assert new_mode in (cs.CS_MODE_ARM, cs.CS_MODE_THUMB)
         self.mode = new_mode
-        self._md.mode = new_mode
+        self.disasm.disasm.mode = new_mode
```

Comment:
No that is proper -> self.disasm is the manticore disassembler (e.g., `core.cpu.disasm.CapstoneDisasm`, self.disasm.disasm is the actual object (`capstone.Cs`), and the mode needs to be updated if we switch.
If this is confusing perhaps naming should be reconsidered?